notifications_utils/template.py

```diff
@@ -620,8 +624,18 @@ def __init__(self, keys):
         super(NoPlaceholderForDataError, self).__init__(", ".join(keys))
 
 
-def get_sms_fragment_count(character_count):
-    return 1 if character_count <= 160 else math.ceil(float(character_count) / 153)
+def get_sms_fragment_count(character_count, unicode):
+    if unicode:
+        return 1 if character_count <= 70 else math.ceil(float(character_count) / 67)
+    else:
+        return 1 if character_count <= 160 else math.ceil(float(character_count) / 153)
+
+
+def is_unicode(content):
+    for char in SanitiseText.WELSH_NON_GSM_CHARACTERS:
+        if char in content:
+            return True
+    return False
```

Comment:
can use set maths to clean make this more efficient, since it'll go only have to loop through `WELSH_NON_GSM_CHARACTERS` and `content` once each to construct the set, and then once through one of the sets to check intersection. (this would reduce complexity from O(n^2) to O(n))

```python
return set(content) & set(SanitiseText.WELSH_NON_GSM_CHARACTERS)
```